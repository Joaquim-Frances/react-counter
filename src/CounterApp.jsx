import React, { useState } from 'react';
import PropTypes from "prop-types";


const CounterApp = ({value = 0}) =>{

    const [ counter, setCounter ]  = useState( 0 );

    const handleAdd = ()=> setCounter( counter + 1);

    const handleReset = ()=>  setCounter( value );

    const handleSubstract = ()=> setCounter( counter - 1 );
      
    return( 
            <div>
                <h1>Counter App</h1>
                <h2> { counter }</h2>
                <button onClick= { handleAdd } >+1</button>
                <button onClick= { handleReset } >Reset</button>
                <button onClick= { handleSubstract } >-1</button>
            </div>
        )

}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
};

export default CounterApp;