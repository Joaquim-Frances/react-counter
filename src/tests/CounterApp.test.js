import React from 'react';
import CounterApp from '../CounterApp';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';


describe('Basic tests from CounterApp', () => {
    
    test('Testin Basic CounterApp HTML Structure', () => {
        
        const value = 100;
        const wrapper = shallow( <CounterApp value={ value } /> );

        expect(wrapper).toMatchSnapshot();

    })
    
    test('Testin CounterApp starting propas value of 100', () => {
        
        const value = 100;
        const wrapper = shallow( <CounterApp value={ value } /> );

        const initialValue =  wrapper.find('h2').text().trim();
        expect(initialValue).toBe('100')

    })


})
