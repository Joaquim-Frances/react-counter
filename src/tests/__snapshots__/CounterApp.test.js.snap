// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic tests from CounterApp Testin Basic CounterApp HTML Structure 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
     
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
